.github/workflows/CI_Tests.yml

```
@@ -29,9 +29,16 @@ jobs:
     - name: Install this package
       run: pip install -e '.[dev]'
     
-    - name: Run pytest
-      run: pytest
+    - name: Run tests and generate coverage reports
+      run: pytest --cov tests/
     
+    - name: Upload coverage reports to codecov
+      uses: codecov/codecov-action@v5
+      with:
+        token: ${{ secrets.CODECOV_TOKEN }}
+        files: coverage.info
+        fail_ci_if_error: false
```

```suggestion
    - name: Upload coverage reports to codecov
      if: runner.os == 'Linux'
      uses: codecov/codecov-action@v5
      with:
        token: ${{ secrets.CODECOV_TOKEN }}
        files: coverage.info
```

We only need to upload the report from one of the operating systems, and we do want to fail the ci if this report fails to upload

---

Actually `fail_ci_if_error: false` ensures that CI doesn't fail because pytests failed